### PR TITLE
Add useGlobalPrefix option

### LINKED ISSUE #142: GraphQL does not respect global path prefix
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

GraphQL does not respect the [global path prefix](https://docs.nestjs.com/faq/global-prefix)

## Expected behavior

GraphQL routes should be mounted at the global path prefix, with an option to opt out of that behavior

## Minimal reproduction of the problem with instructions

1. Follow instructions for setting up a simple [GraphQL server with Nestjs](https://docs.nestjs.com/graphql/quick-start).
2. Set the [global prefix](https://docs.nestjs.com/faq/global-prefix) to something like `v1`
3. Expect to see GraphQL available at `/v1/graphql` but it is still only available at `/graphql`

## What is the motivation / use case for changing the behavior?

With the name "global prefix" the expectation is that any routes that are mounted should also be prefixed - not just controllers.

## Environment

```
"@nestjs/common": "5.7.3",
"@nestjs/core": "5.7.3",
"@nestjs/graphql": "5.5.4",
```

### PR DESCRIPTION
**Pull request by willsoto**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: Closes #142 

## What is the new behavior?

Adds a new option called `useGlobalPrefix?: boolean` which is available on `GqlModuleOptions`

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

Implementation inspired by https://github.com/nestjs/terminus/pull/261

### CODE REVIEW

**Comment by kamilmysliwiec** [comment]
Great job! Thank you

File: tests/e2e/global-prefix.spec.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+describe('GraphQL (global prefix)', () => {
+  let app: INestApplication;
+
+  beforeEach(async () => {
+    const module = await Test.createTestingModule({
+      imports: [GlobalPrefixModule],
+    }).compile();
+
+    app = module.createNestApplication();
+    app.setGlobalPrefix('/api/v1');
```
Comment (meteorlxy) [inline tests/e2e/global-prefix.spec.ts]: This option will fail without the starting slash, i.e. `app.setGlobalPrefix('api/v1');`

Comment (johannesschobel) [reply tests/e2e/global-prefix.spec.ts]: that is indeed a problem! can this be fixed? the official docs ( https://docs.nestjs.com/faq/global-prefix ) indicates the `globalPrefix` **without** the `/` at the start / end..

Comment (meteorlxy) [reply tests/e2e/global-prefix.spec.ts]: @johannesschobel

See #364

Comment (johannesschobel) [reply tests/e2e/global-prefix.spec.ts]: haha.. yeah, i was just about to make a PR but then i found yours..

**Comment by johannesschobel** [comment]
This PR also does not work with `GraphQlModule.forRootAsync()` but only for the `.forRoot()`. Can you confirm this @meteorlxy ?

**Comment by meteorlxy** [comment]
@johannesschobel 

It works well with `GraphQlModule.forRootAsync()` in my code.

image

**Comment by johannesschobel** [comment]
thanks, i will try.. i have not explicitly set the `path` attribute.. maybe that's the issue?!

**Comment by johannesschobel** [comment]
ok... with the `path` attribute that has an explicit `/` at the start it works..

**Pull request merged**